### PR TITLE
Added Organization.CreateOrUpdate

### PR DESCRIPTION
**Pull request by MelGrubb**
I added the missing methods and related tests.

### CODE REVIEW

File: test/ZendeskApi_v2.Test/OrganizationTests.cs [hunk 1]
```diff
@@ -135,6 +135,42 @@ public void CanCreateUpdateAndDeleteOrganizations()
...
+            var res = await api.Organizations.CreateOrUpdateOrganizationAsync(new Organization()
+            {
+                Name = "Test Org (original)"
+            });
+
+            Assert.Greater(res.Organization.Id, 0);
+
+            res.Organization.Name = "Test Org (updated)";
+            var update = await api.Organizations.CreateOrUpdateOrganizationAsync(res.Organization);
+            Assert.AreEqual(update.Organization.Id, res.Organization.Id);
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/OrganizationTests.cs:168]:
```csharp
            Assert.That(update.Organization.Id, Is.EqualTo(res.Organization.Id));
            Assert.That(update.Organization.Name,  Is.EqualTo(res.Organization.Name));

            Assert.That( await api.Organizations.DeleteOrganizationAsync(res.Organization.Id.Value), Is.True);
```

File: test/ZendeskApi_v2.Test/OrganizationTests.cs [hunk 2]
```diff
@@ -135,6 +135,42 @@ public void CanCreateUpdateAndDeleteOrganizations()
...
+
+        [Test]
+        public async Task CanCreateOrUpdateOrganizationsAsync()
+        {
+            var res = await api.Organizations.CreateOrUpdateOrganizationAsync(new Organization()
+            {
+                Name = "Test Org (original)"
+            });
+
+            Assert.Greater(res.Organization.Id, 0);
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/OrganizationTests.cs:164]:
```csharp
Assert.That(res.Organization.Id, Is.GreaterThan(0));
```

**Review by mozts2005 — COMMENTED** [review]
please, have a look at the code changes.

File: test/ZendeskApi_v2.Test/OrganizationTests.cs [hunk 3]
```diff
@@ -135,6 +135,42 @@ public void CanCreateUpdateAndDeleteOrganizations()
...
 
+        [Test]
+        public void CanCreateOrUpdateOrganizations()
+        {
+            var res = api.Organizations.CreateOrUpdateOrganization(new Organization()
+            {
+                Name = "Test Org (original)"
+            });
+
+            Assert.Greater(res.Organization.Id, 0);
```
[the remark this answers is not in the document]
Comment (MelGrubb) [reply test/ZendeskApi_v2.Test/OrganizationTests.cs:146]: These tests are direct copies of the style of other existing tests in the same and other files. Isn't consistency better?

[the remark this answers is not in the document]
Comment (MelGrubb) [reply test/ZendeskApi_v2.Test/OrganizationTests.cs:146]: Also, you should really look at an assertion library like Shouldly. It makes the code AND the error messages easier to read.

[the remark this answers is not in the document]
Comment (mozts2005) [reply test/ZendeskApi_v2.Test/OrganizationTests.cs:146]: I am working on changing all test cases to this format. When using the ```Assert.That``` syntax the error messages are the same style that Shouldly uses with out the need for a separate dependency.

File: test/ZendeskApi_v2.Test/OrganizationTests.cs [hunk 4]
```diff
@@ -135,6 +135,42 @@ public void CanCreateUpdateAndDeleteOrganizations()
...
+
+            res.Organization.Name = "Test Org (updated)";
+            var update = await api.Organizations.CreateOrUpdateOrganizationAsync(res.Organization);
+            Assert.AreEqual(update.Organization.Id, res.Organization.Id);
+            Assert.AreEqual(update.Organization.Name, res.Organization.Name);
+
+            Assert.True(api.Organizations.DeleteOrganization(res.Organization.Id.Value));
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/OrganizationTests.cs:171]:
```suggestion
            Assert.That(update.Organization.Id, Is.EqualTo(res.Organization.Id));
            Assert.That(update.Organization.Name,  Is.EqualTo(res.Organization.Name));

            Assert.That( await api.Organizations.DeleteOrganizationAsync(res.Organization.Id.Value), Is.True);
```

File: test/ZendeskApi_v2.Test/OrganizationTests.cs [hunk 5]
```diff
@@ -135,6 +135,42 @@ public void CanCreateUpdateAndDeleteOrganizations()
...
+            });
+
+            Assert.Greater(res.Organization.Id, 0);
+
+            res.Organization.Name = "Test Org (updated)";
+            var update = api.Organizations.CreateOrUpdateOrganization(res.Organization);
+            Assert.AreEqual(update.Organization.Id, res.Organization.Id);
+            Assert.AreEqual(update.Organization.Name, res.Organization.Name);
+
+            Assert.True(api.Organizations.DeleteOrganization(res.Organization.Id.Value));
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/OrganizationTests.cs:153]:
```suggestion
            Assert.That(update.Organization.Id, Is.EqualTo(res.Organization.Id));
            Assert.That(update.Organization.Name,  Is.EqualTo(res.Organization.Name));

            Assert.That(api.Organizations.DeleteOrganization(res.Organization.Id.Value), Is.True);
```

**Review by mozts2005 — CHANGES_REQUESTED** [review]

File: test/ZendeskApi_v2.Test/OrganizationTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (MelGrubb) [reply test/ZendeskApi_v2.Test/OrganizationTests.cs:146]: Does this hold for the other assertions as well?
```
Assert.That(update.Organization, Is.EqualTo(res.Organization.Id));
Assert.That(update.Organization.Name, Is.EqualTo(res.Organization.Name));
Assert.That(api.Organizations.DeleteOrganization(res.Organization.Id.Value), Is.True);
```

[the remark this answers is not in the document]
Comment (MelGrubb) [reply test/ZendeskApi_v2.Test/OrganizationTests.cs]: Asserts have been updated.

You should be able to make a change everywhere in one clean sweep with some regex search and replace. I've certainly swapped between MSTest and NUnit that way before. That would keep the code in an internally-consistent state by changing them all at once. I'm not saying they're FUN to write, but you spend 15-30 minutes wrestling with RegEx and then watch the magic happen everywhere all at once. Satisfying.

**Review by mozts2005 — APPROVED** [review]

**Pull request merged**